src/hotspot/cpu/aarch64/universalUpcallHandler_aarch64.cpp

```diff
@@ -71,7 +71,7 @@ address ProgrammableUpcallHandler::generate_upcall_stub(jobject rec, jobject jab
 
   // Capture prev stack pointer (stack arguments base)
   __ add(rscratch1, rfp, 16);   // Skip saved FP and LR
-  __ str(rscratch1, Address(sp, layout.stack_args));
+  __ str(rscratch1, Address(sp, (int)layout.stack_args));  // x86 casts to int also
```

Comment:
```suggestion
  __ str(rscratch1, Address(sp, checked_cast<int>(layout.stack_args)));  // x86 casts to int also
```

Reply:
... because this is UB if it doesn't fit.

Reply:
Yes, I changed this and wasn't really sure how to fix it better.  I'd love if Address had a unit64_t argument but all the callers that just passed 0 wouldn't compile.  The code is like that in the x86 version. So I thought that'd be safe.

Reply:
Well, I'm not convinced the x86 version is safe, really. It probably is, but the checked_cast costs nothing in production.

Reply:
Actually, thinking some more, I can see no guarantee that this `Address` is valid either. So I'm going to change my mind.

Reply:
I'm going to remove the cast or cast it to uint64_t in universalUpcallHandler. thanks for bearing with me!